Tasks/DownloadBuildArtifacts/main.ts

```diff
@@ -0,0 +1,139 @@
+var path = require('path')
+var url = require('url')
+
+import * as tl from 'vsts-task-lib/task';
+import { IBuildApi } from 'vso-node-api/BuildApi';
+import { IRequestHandler } from 'vso-node-api/interfaces/common/VsoBaseInterfaces';
+import { WebApi, getHandlerFromToken } from 'vso-node-api/WebApi';
+
+import * as models from 'item-level-downloader/Models';
+import * as engine from 'item-level-downloader/Engine';
+import * as providers from 'item-level-downloader/Providers';
+import * as webHandlers from 'item-level-downloader/Providers/Handlers';
+
+tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+async function main(): Promise<void> {
+    var promise = new Promise<void>(async (resolve, reject) => {
+        var buildType: string = tl.getInput("buildType", true);
+        var isCurrentBuild : boolean = buildType.toLowerCase() === 'current';
+        var projectId : string =  isCurrentBuild ? tl.getVariable("System.TeamProjectId") : tl.getInput("project", true);
+        var definitionId : string = isCurrentBuild ? '' : tl.getInput("definition", true);
+        var buildId : number = parseInt(isCurrentBuild ? tl.getVariable("Build.BuildId") : tl.getInput("buildId", true));
+        var downloadPath : string = tl.getInput("downloadPath", true);
+        var downloadType : string = tl.getInput("downloadType", true);
+
+        var endpointUrl : string = tl.getVariable("System.TeamFoundationCollectionUri");
+        var accessToken : string = tl.getEndpointAuthorizationParameter('SYSTEMVSSCONNECTION', 'AccessToken', false);
+        var credentialHandler : IRequestHandler = getHandlerFromToken(accessToken);
+        var webApi: WebApi = new WebApi(endpointUrl, credentialHandler);
+        var debugMode : string = tl.getVariable('System.Debug');
+        var isVerbose : boolean = debugMode ? debugMode.toLowerCase() != 'false' : false;
+        var parallelLimit : number = +tl.getVariable("release.artifact.download.parallellimit");
+
+        var templatePath: string = path.join(__dirname, 'vsts.handlebars.txt');
+        var buildApi : IBuildApi = webApi.getBuildApi();
+        var artifacts = [];
+        var itemPattern : string = '**';
+
+        // verfiy that buildId belongs to the definition selected
+        if (definitionId) {
+            var builds = await buildApi.getBuilds(projectId, [parseInt(definitionId)]).catch((reason) => {
```

Comment:
As written, this will get up to 1000 builds.  Way too many.  Why don't you query for the specific build and verify that its definition id matches?

Reply:
Good point! i have changed the logic to just fetch the build and compare definition id.